### PR TITLE
Hide incompatible index patterns

### PR DESCRIPTION
**Pull request by Kapian1234**
### Description
before:
Screenshot 2024-10-17 at 17 13 59
Screenshot 2024-10-17 at 17 14 15

after:
Screenshot 2024-10-17 at 17 12 52
Screenshot 2024-10-17 at 17 13 11

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test.
- [ ] New functionality has user manual doc added.
- [ ] Commits are signed per the DCO using --signoff.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/visualization/source_selector.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 import React, { useCallback, useMemo, useState, useEffect, useRef } from 'react';
 import { i18n } from '@osd/i18n';
-
+import { EuiComboBoxOptionOption } from '@elastic/eui';
```
Comment (ruanyl) [inline public/components/visualization/source_selector.tsx:8]: This new import seems is not used?

Comment (Kapian1234) [reply public/components/visualization/source_selector.tsx:8]: yes I'll delete it

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,11 +3,12 @@
...
 
 ### Unreleased
-- fix: make sure $schema always added to LLM generated vega json object([252](https://github.com/opensearch-project/dashboards-assistant/pull/252))
+
+- fix: make sure \$schema always added to LLM generated vega json object([252](https://github.com/opensearch-project/dashboards-assistant/pull/252))
 - feat: expose a general function for agent execution([268](https://github.com/opensearch-project/dashboards-assistant/pull/268))
 - Fix CVE-2024-4067 ([#269](https://github.com/opensearch-project/dashboards-assistant/pull/269))
 - feat: add a dashboards-assistant trigger in query editor([265](https://github.com/opensearch-project/dashboards-assistant/pull/265))
-- fix: make sure $schema always added to LLM generated vega json object([#252](https://github.com/opensearch-project/dashboards-assistant/pull/252))
+- fix: make sure \$schema always added to LLM generated vega json object([#252](https://github.com/opensearch-project/dashboards-assistant/pull/252))
```
Comment (ruanyl) [inline CHANGELOG.md:11]: Let's revert unrelated changes

File: CHANGELOG.md [hunk 3]
```diff
@@ -52,3 +52,4 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Refactor popover to add message action bar and add metrics to thumb-up and thumb-down([#304](https://github.com/opensearch-project/dashboards-assistant/pull/304))
 - Support log pattern analysis ([#339](https://github.com/opensearch-project/dashboards-assistant/pull/339))
 - Pass top N log pattern data from alerting to agent ([#341] (https://github.com/opensearch-project/dashboards-assistant/pull/341))
+- Hide incompatible index patterns ([#354] (https://github.com/opensearch-project/dashboards-assistant/pull/354))
```
Comment (ruanyl) [inline CHANGELOG.md:55]: move this to `Unreleased`

File: public/components/visualization/source_selector.tsx [hunk 4]
```diff
@@ -171,10 +166,27 @@ export const SourceSelector = ({
...
+  const options = useMemo(() => {
+    if (dataSourceOptions[0] && dataSourceOptions[0].options.length > 0) {
+      return dataSourceOptions;
+    }
+    return [
+      {
+        label: 'Index patterns',
+        options: [
+          {
+            label: 'No supported index patterns',
```
Comment (Hailong-am) [inline public/components/visualization/source_selector.tsx:178]: i18n

Comment (Kapian1234) [reply public/components/visualization/source_selector.tsx:178]: sorry I forgot it

File: CHANGELOG.md (hunk 3 above)
Comment (Kapian1234) [reply CHANGELOG.md:55]: ok

**Review by ruanyl — APPROVED** [review]

**Pull request merged**